sound/soc/sof/intel/hda.c

```diff
@@ -1025,7 +1025,7 @@ static bool link_slaves_found(struct snd_sof_dev *sdev,
 	struct sdw_intel_slave_id *ids = sdw->ids;
 	int num_slaves = sdw->num_slaves;
 	unsigned int part_id, link_id, unique_id, mfg_id;
-	int i, j;
+	int i, j, k;
```

Comment:
suggested edit on commit message:

ASoC: SOF: Intel: check number of identical SoundWire devices per link.

The existing link_slaves_found() function will still return true if the link descriptor exposes more than one peripheral with the same mfg_id/part_id, but only one peripheral is present in hardware.

The patch double-checks if there is any other peripheral with the same part id
in cases where a single one was expected.